crates/dojo-project/src/lib.rs

```diff
@@ -84,3 +91,19 @@ impl From<ProjectConfig> for cairo_lang_project::ProjectConfig {
         }
     }
 }
+
+pub fn read_metadata(path: Option<&Utf8Path>) -> Result<ProjectMetadata> {
+    let manifest_path = ops::find_manifest_path(path).unwrap();
+
+    let config = Config::builder(manifest_path)
+        .ui_verbosity(Verbosity::Verbose)
+        .log_filter_directive(env::var_os("SCARB_LOG"))
+        .build()
+        .unwrap();
+
+    let ws = ops::read_workspace(config.manifest_path(), &config).unwrap();
```

Comment:
Note: For you this might make sense here to embed Scarb as library, but for language server I think it definitely look for Scarb binary so that it's not tightly coupled with particular versions of dependencies.